File: tests/fixtures/c_simple/service.c
#include "service.h"
#include <string.h>

#define MAX_ITEMS 100

static Item store[MAX_ITEMS];
static int count = 0;

const char* get_item(int id) {
    for (int i = 0; i < count; i++) {
        if (store[i].id == id && store[i].active) {
            return store[i].name;
        }
    }
    return "";
}

int create_item(const char* name) {
    if (count >= MAX_ITEMS) return -1;
    count++;
    store[count - 1].id = count;
    strncpy(store[count - 1].name, name, 255);
    store[count - 1].name[255] = '\0';
    store[count - 1].active = 1;
    return count;
}

void delete_item(int id) {
    for (int i = 0; i < count; i++) {
        if (store[i].id == id) {
            store[i].active = 0;
            return;
        }
    }
}

int update_item(int id, const char* name) {
    for (int i = 0; i < count; i++) {
        if (store[i].id == id && store[i].active) {
            strncpy(store[i].name, name, 255);
            store[i].name[255] = '\0';
            return 1;
        }
    }
    return 0;
}

int list_items(Item* buffer, int max_count) {
    int found = 0;
    for (int i = 0; i < count && found < max_count; i++) {
        if (store[i].active) {
            buffer[found] = store[i];
            found++;
        }
    }
    return found;
}

int item_count(void) {
    int active = 0;
    for (int i = 0; i < count; i++) {
        if (store[i].active) active++;
    }
    return active;
}

File: tests/fixtures/go_simple/service.go
package service

import "myapp/model"

type DataService struct {
	store map[int]model.Item
	count int
}

type ItemRecord struct {
	Id   int
	Name string
}

func NewDataService() *DataService {
	return &DataService{
		store: make(map[int]model.Item),
	}
}

func (s *DataService) GetItem(id int) string {
	item, ok := s.store[id]
	if !ok {
		return ""
	}
	return item.Name
}

func (s *DataService) CreateItem(name string) int {
	s.count++
	s.store[s.count] = model.Item{
		Id:     s.count,
		Name:   name,
		Active: true,
	}
	return s.count
}

func (s *DataService) DeleteItem(id int) bool {
	if _, ok := s.store[id]; !ok {
		return false
	}
	delete(s.store, id)
	return true
}

func (s *DataService) ListItems() []ItemRecord {
	records := make([]ItemRecord, 0, len(s.store))
	for _, item := range s.store {
		records = append(records, ItemRecord{Id: item.Id, Name: item.Name})
	}
	return records
}

func (s *DataService) UpdateItem(id int, name string) bool {
	item, ok := s.store[id]
	if !ok {
		return false
	}
	item.Name = name
	s.store[id] = item
	return true
}
